drivers/gpio/Kconfig

```diff
@@ -16,6 +16,7 @@ module-str = gpio
 source "subsys/logging/Kconfig.template.log_config"
 
 config GPIO_SHELL
+	depends on SHELL
```

Comment:
Please move this after the `bool` line, to be consistent with most other Kconfig declarations, e.g. https://github.com/zephyrproject-rtos/zephyr/blob/966015f503d1438c25d59793762495452be5ebbc/drivers/i2c/Kconfig#L19